presto-hive/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -258,55 +262,55 @@ public void flushCache()
     }
 
     @Override
-    public Map<String, HiveColumnStatistics> getTableColumnStatistics(String databaseName, String tableName)
+    public PartitionStatistics getTableStatistics(String databaseName, String tableName)
     {
-        return get(tableColumnStatisticsCache, new HiveTableName(databaseName, tableName));
+        return get(tableStatisticsCache, new HiveTableName(databaseName, tableName));
     }
 
-    private Map<String, HiveColumnStatistics> loadTableColumnStatistics(HiveTableName hiveTableName)
+    private PartitionStatistics loadTableColumnStatistics(HiveTableName hiveTableName)
     {
-        return delegate.getTableColumnStatistics(hiveTableName.getDatabaseName(), hiveTableName.getTableName());
+        return delegate.getTableStatistics(hiveTableName.getDatabaseName(), hiveTableName.getTableName());
     }
 
     @Override
-    public Map<String, Map<String, HiveColumnStatistics>> getPartitionColumnStatistics(String databaseName, String tableName, Set<String> partitionNames)
+    public Map<String, PartitionStatistics> getPartitionStatistics(String databaseName, String tableName, Set<String> partitionNames)
     {
         List<HivePartitionName> partitions = partitionNames.stream()
                 .map(partitionName -> HivePartitionName.partition(databaseName, tableName, partitionName))
                 .collect(toImmutableList());
-        Map<HivePartitionName, Map<String, HiveColumnStatistics>> statistics = getAll(partitionColumnStatisticsCache, partitions);
+        Map<HivePartitionName, PartitionStatistics> statistics = getAll(partitionStatisticsCache, partitions);
         return statistics.entrySet()
                 .stream()
-                .filter(entry -> !entry.getValue().isEmpty())
                 .collect(toImmutableMap(entry -> entry.getKey().getPartitionName(), Entry::getValue));
     }
 
-    private Map<String, HiveColumnStatistics> loadPartitionColumnStatistics(HivePartitionName partition)
+    private PartitionStatistics loadPartitionColumnStatistics(HivePartitionName partition)
     {
-        Map<String, Map<String, HiveColumnStatistics>> columnStatistics = delegate.getPartitionColumnStatistics(
+        Map<String, PartitionStatistics> partitionStatistics = delegate.getPartitionStatistics(
                 partition.getHiveTableName().getDatabaseName(),
                 partition.getHiveTableName().getTableName(),
                 ImmutableSet.of(partition.getPartitionName()));
-        return columnStatistics.getOrDefault(partition.getPartitionName(), ImmutableMap.of());
+        if (!partitionStatistics.containsKey(partition.getPartitionName())) {
+            throw new PrestoException(HIVE_PARTITION_DROPPED_DURING_QUERY, "Statistics result does not contain entry for partition: " + partition.getPartitionName());
```

Comment:
How does it change the semantic?

Reply:
Ok. I think i understand your question now. So before this commit this method used to return only the partition *column* statistics, which are allowed to be missing for a give partition. But now the contract has changed. If the statistics are missing, the underlying method should always return a non-null `PartitionStatistics` object for every partition requested. If the statistics are missing - it will have an empty map inside. The underlying metastore can return `null` only if the partition is no longer there -> the partition has been dropped during the execution.

Reply:
It would be nice to separate semantic changes from refactorings, but I understand that managing such big PRs could be a pain.